code/modules/reagents/chemistry/reagents/drink_reagents.dm

```diff
@@ -661,9 +661,13 @@
 	desc = "Surprisingly it isn't grey."
 	icon_state = "grey_bull_glass"
 
-/datum/reagent/consumable/grey_bull/on_mob_metabolize(mob/living/affected_mob)
+/datum/reagent/consumable/grey_bull/on_mob_metabolize(mob/living/carbon/affected_atom)
 	..()
-	ADD_TRAIT(affected_mob, TRAIT_SHOCKIMMUNE, type)
+	ADD_TRAIT(affected_atom, TRAIT_SHOCKIMMUNE, type)
+	var/obj/item/organ/internal/liver/liver = affected_atom.get_organ_slot(ORGAN_SLOT_LIVER)
+	if(HAS_TRAIT(liver, TRAIT_MAINTENANCE_METABOLISM))
+		affected_atom.add_mood_event("maintenance_fun", /datum/mood_event/maintenance_high)
+		metabolization_rate *= 0.8
```

Comment:
wont this scale every metab iteration? The metab rate progressively gets lowered.

so first metab will be 80%, then the next metab will be 80% of 80%, third 80% of 80% of 80%...

Reply:
nope ur good